python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py

```diff
@@ -204,6 +212,14 @@ def handle_run_event(self, run_id: str, event: DagsterEvent) -> None:
                 )
             )
 
+        if event.event_type == DagsterEventType.PIPELINE_FAILURE and isinstance(
+            event.event_specific_data, JobFailureData
+        ):
+            failure_reason = event.event_specific_data.failure_reason
+
+            if failure_reason and failure_reason != RunFailureReason.UNKNOWN:
+                self.add_run_tags(run_id, {RUN_FAILURE_REASON_TAG: failure_reason.value})
```

Comment:
do we already call `add_run_tags` for every run on some path? just making sure this isnt going to be substantial new load.

Reply:
I think its called a couple of different times during the launch flow already - once at submit time, once at enqueue time at least

Reply:
if we're worried we could potentially gate it on the new run retries flag I am planning to add that ingests it

Reply:
I couldn't remember if it was just in the local grpc case or if it happened in other contexts as well

i think you are right and this should be fine